core/metrics/impl/exposer_impl.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "exposer_impl.hpp"
+#include <thread>
+#include "session_impl.hpp"
+
+namespace kagome::metrics {
+  ExposerImpl::ExposerImpl(
+      const std::shared_ptr<application::AppStateManager> &app_state_manager,
+      Exposer::Configuration exposer_config,
+      Session::Configuration session_config)
+      : logger_{log::createLogger("OpenMetrics", "metrics")},
+        context_{std::make_shared<Context>()},
+        config_{std::move(exposer_config)},
+        session_config_{session_config} {
+    BOOST_ASSERT(app_state_manager);
+    app_state_manager->takeControl(*this);
+  }
+
+  bool ExposerImpl::prepare() {
+    try {
+      acceptor_ = std::make_unique<Acceptor>(*context_, config_.endpoint);
+    } catch (const boost::wrapexcept<boost::system::system_error> &exception) {
+      logger_->critical("Failed to prepare a listener: {}", exception.what());
```

Comment:
Please use SL_CRITICAL macros
```suggestion
      SL_CRITICAL(logger_, "Failed to prepare a listener: {}", exception.what());
```

Change in other places as well